### PR TITLE
[Refactor] Refactor array_concat to support all arrays

### PR DESCRIPTION
**Pull request by alvin-coding**
## What type of PR is this：
- [ ] BugFix
- [ ] Feature
- [ ] Enhancement
- [X] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes #

## Problem Summary(Required) ：

In this CL, I refactor the implementation of array_concat to support all kinds of arrays

```
mysql> select array_concat([map_from_arrays([1], [2])], [map_from_arrays([2], [3])]);
+------------------------------------------------------------------------+
| array_concat([map_from_arrays([1], [2])], [map_from_arrays([2], [3])]) |
+------------------------------------------------------------------------+
| [{1:2},{2:3}]                                                          |
+------------------------------------------------------------------------+
1 row in set (0.01 sec)
```

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [ ] I have checked the version labels which the pr will be auto backported to target branch
  - [ ] 2.5
  - [ ] 2.4
  - [ ] 2.3
  - [ ] 2.2

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by alvin-coding** [comment]
run starrocks_admit_test

File: be/src/exprs/array_functions.cpp [hunk 1]
```diff
@@ -1513,4 +1513,70 @@ StatusOr<ColumnPtr> ArrayFunctions::array_max_varchar([[maybe_unused]] FunctionC
...
+StatusOr<ColumnPtr> ArrayFunctions::concat(FunctionContext* ctx, const Columns& columns) {
+    DCHECK(columns.size() > 1);
+    RETURN_IF_COLUMNS_ONLY_NULL(columns);
+
+    auto num_rows = columns[0]->size();
+
+    // compute nulls
+    NullColumnPtr nulls;
+    for (auto& column : columns) {
+        if (column->is_nullable()) {
```
[the remark this answers is not in the document]
Comment (alvin-coding) [reply be/src/exprs/array_functions.cpp:1525]: Good suggestion!

**Review by murphyatwork — DISMISSED** [review]

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]


:heart_eyes: **pass** : 0 / 0 (0%)

**Review by imay — APPROVED** [review]

**Pull request merged**